clients/ordering/src/components/checkout.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { Component } from 'react';
+import {
+  List,
+  ListItem,
+  ListItemText,
+  Paper,
+  Typography,
+  Grid,
+  TextField,
+  Button,
+  Theme,
+  withStyles,
+  createStyles,
+  WithStyles,
+} from '@material-ui/core';
+import Slider from '@material-ui/lab/Slider';
+import { Elements } from 'react-stripe-elements';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import CardForm from './cardForm';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    button: {
+      whiteSpace: 'nowrap',
+      border: 0,
+      outline: 0,
+      display: 'inline-block',
+      height: '40px',
+      lineHeight: '40px',
+      padding: '0 14px',
+      boxShadow:
+        '0 4px 6px rgba(50, 50, 93, .11), 0 1px 3px rgba(0, 0, 0, .08)',
+      color: '#fff',
+      borderRadius: '4px',
+      fontSize: '15px',
+      fontWeight: 600,
+      textTransform: 'uppercase',
+      letterSpacing: '0.025em',
+      backgroundColor: '#6772e5',
+      textDecoration: 'none',
+      WebkitTransition: 'all 150ms ease',
+      transition: 'all 150ms ease',
+      marginTop: '10px',
+      marginLeft: '20px',
+    },
+    paper: {
+      padding: 25,
+      marginLeft: 550,
+      marginRight: 550,
+      marginBottom: 25,
+      align: 'center',
+    },
+    slider: {
+      marginLeft: 40,
+    },
+    totalText: {
+      fontWeight: 'bold',
+      fontSize: 'large',
+    },
+  });
+
+interface CheckoutProps extends WithStyles<typeof styles> {
+  addedItems: any[];
+  boughtItems: any[];
+  total: number;
+  finalTotal: number;
+}
+
+class Checkout extends Component<CheckoutProps> {
+  state = {
+    value: 15,
```

Comment:
Add types for the component's local state, so something like `Component<CheckoutProps, CheckoutState>`. Preferably don't make the type `any` if possible.